runtime/compiler/runtime/JITServerAOTCache.cpp

```diff
@@ -58,13 +64,46 @@ AOTCacheClassLoaderRecord::AOTCacheClassLoaderRecord(uintptr_t id, const uint8_t
    {
    }
 
+AOTCacheClassLoaderRecord::AOTCacheClassLoaderRecord() :
+   _data()
+   {
+   }
+
 AOTCacheClassLoaderRecord *
 AOTCacheClassLoaderRecord::create(uintptr_t id, const uint8_t *name, size_t nameLength)
    {
    void *ptr = AOTCacheRecord::allocate(size(nameLength));
    return new (ptr) AOTCacheClassLoaderRecord(id, name, nameLength);
    }
 
+AOTCacheClassLoaderRecord *
+AOTCacheClassLoaderRecord::create(size_t nameLength)
+   {
+   void *ptr = AOTCacheRecord::allocate(size(nameLength));
+   return new (ptr) AOTCacheClassLoaderRecord();
+   }
+
+AOTCacheClassLoaderRecord *
+AOTCacheClassLoaderRecord::read(FILE *f)
+   {
+   ClassLoaderSerializationRecord header;
+   if (1 != fread(&header, sizeof(header), 1, f))
+      return NULL;
+
+   if ((AOTSerializationRecordType::ClassLoader != header.type()) ||
```

Comment:
I think it's worth moving this logic to `isValid()` methods in each serialization record type (with the common part in `AOTSerializationRecord::isValid()`).